### PR TITLE
regen code to use ManagementException

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
Depends on https://github.com/Azure/azure-sdk-for-java/pull/10822

```
export SPEC_ROOT=/c/github/azure-rest-api-specs
export AUTOREST_JAVA=/c/github_fork/autorest.java

gulp codegen --projects features --spec-root $SPEC_ROOT --autorest-java $AUTOREST_JAVA
gulp codegen --projects policy --spec-root $SPEC_ROOT --autorest-java $AUTOREST_JAVA --preserve
gulp codegen --projects subscriptions --spec-root $SPEC_ROOT --autorest-java $AUTOREST_JAVA --preserve
gulp codegen --projects resources --spec-root $SPEC_ROOT --autorest-java $AUTOREST_JAVA --preserve

gulp codegen --projects storage --spec-root $SPEC_ROOT --autorest-java $AUTOREST_JAVA

gulp codegen --projects arm-keyvault --spec-root $SPEC_ROOT --autorest-java $AUTOREST_JAVA

gulp codegen --projects graphrbac --spec-root $SPEC_ROOT --autorest-java $AUTOREST_JAVA
gulp codegen --projects authorization --spec-root $SPEC_ROOT --autorest-java $AUTOREST_JAVA --preserve

gulp codegen --projects msi --spec-root $SPEC_ROOT --autorest-java $AUTOREST_JAVA

gulp codegen --projects network --spec-root $SPEC_ROOT --autorest-java $AUTOREST_JAVA
gulp codegen --projects compute --spec-root $SPEC_ROOT --autorest-java $AUTOREST_JAVA
gulp codegen --projects appservice --spec-root $SPEC_ROOT --autorest-java $AUTOREST_JAVA
gulp codegen --projects containerregistry --spec-root $SPEC_ROOT --autorest-java $AUTOREST_JAVA
gulp codegen --projects containerservice --spec-root $SPEC_ROOT --autorest-java $AUTOREST_JAVA
gulp codegen --projects cosmosdb --spec-root $SPEC_ROOT --autorest-java $AUTOREST_JAVA
gulp codegen --projects dns --spec-root $SPEC_ROOT --autorest-java $AUTOREST_JAVA
gulp codegen --projects monitor --spec-root $SPEC_ROOT --autorest-java $AUTOREST_JAVA
gulp codegen --projects sql --spec-root $SPEC_ROOT --autorest-java $AUTOREST_JAVA
```

### CODE REVIEW

File: sdk/compute/mgmt/src/main/java/com/azure/management/compute/implementation/GalleryImageVersionImpl.java [hunk 1]
```diff
@@ -201,13 +201,13 @@ public GalleryImageVersionImpl withLocation(String location) {
 
     @Override
     public GalleryImageVersionImpl withSourceCustomImage(String customImageId) {
```
Comment (weidongxu-microsoft) [inline sdk/compute/mgmt/src/main/java/com/azure/management/compute/implementation/GalleryImageVersionImpl.java:203]: service swagger changed again. temporary comment

File: sdk/appservice/mgmt/src/main/java/com/azure/management/appservice/implementation/WebAppMsiHandler.java [hunk 2]
```diff
@@ -120,7 +120,7 @@ WebAppMsiHandler withExistingExternalManagedServiceIdentity(Identity identity) {
      * @return WebAppMsiHandler
      */
     WebAppMsiHandler withoutExternalManagedServiceIdentity(String identityId) {
-        this.userAssignedIdentities.put(identityId, null);
+        this.userAssignedIdentities.remove(identityId);
```
Comment (weidongxu-microsoft) [inline sdk/appservice/mgmt/src/main/java/com/azure/management/appservice/implementation/WebAppMsiHandler.java:123]: @xccc-msft Please let me know why it putting `null` to map value.

This causes NPE during client side validation.

Comment (xccc-msft) [reply sdk/appservice/mgmt/src/main/java/com/azure/management/appservice/implementation/WebAppMsiHandler.java]: Just update for tracking purpose: This is by design behavior as the server need to verify null as value in the map to understand it requests to remove the `identityId`.

**Comment by weidongxu-microsoft** [comment]
@ChenTanyi @xccc-msft 
Let me know if you are OK with it. Basically it use a different core-management with ManagementException. And with other fixes on NM_CONFUSE from generator.

Since there going to be more change on core-management pending, I will currently use my maven repo instead of official snapshot repo.

@xccc-msft 
After, please take a look on commented part of `GalleryImageVersionImpl`. I prefer either fix the swagger, or fix the code.

**Comment by weidongxu-microsoft** [comment]
@ChenTanyi 

No, not yet, maybe next week (or we just publish an official beta.8). There is still pending changes (like whether to change to azure-management-core :-) ), so I would temporary update it in my own maven repo.

**Pull request merged**